Utils/FaviconFetcher.swift

```diff
@@ -99,74 +99,75 @@ public class FaviconFetcher: NSObject, NSXMLParserDelegate {
-                        let content = meta.attribute("content"),
-                        let index = content.rangeOfString("URL="),
-                        let url = NSURL(string: content.substringFromIndex(index.startIndex.advancedBy(4))) {
-                            reloadUrl = url
-                    }
+            guard let data = result.successValue where result.isSuccess,
+                let root = try? HTMLDocument(data: data) else {
+                    return deferMaybe([])
+            }
+            var reloadUrl: NSURL? = nil
+            for meta in root.xpath("//head/meta") {
+                if let refresh = meta["http-equiv"] where refresh == "Refresh",
+                    let content = meta["content"],
+                    let index = content.rangeOfString("URL="),
+                    let url = NSURL(string: content.substringFromIndex(index.startIndex.advancedBy(4))) {
+                    reloadUrl = url
                 }
+            }
 
-                if let url = reloadUrl {
-                    return self.parseHTMLForFavicons(url)
-                }
+            if let url = reloadUrl {
+                return self.parseHTMLForFavicons(url)
+            }
 
-                var bestType = IconType.NoneFound
-                element.iterateWithRootXPath("//head//link[contains(@rel, 'icon')]") { link in
-                    var iconType: IconType? = nil
-                    if let rel = link.attribute("rel") {
-                        switch (rel) {
-                            case "shortcut icon":
-                                iconType = .Icon
-                            case "icon":
-                                iconType = .Icon
-                            case "apple-touch-icon":
-                                iconType = .AppleIcon
-                            case "apple-touch-icon-precomposed":
-                                iconType = .AppleIconPrecomposed
-                            default:
-                                iconType = nil
-                        }
+            var bestType = IconType.NoneFound
+            for link in root.xpath("//head//link[contains(@rel, 'icon')]") {
+                var iconType: IconType? = nil
+                if let rel = link["rel"] {
+                    switch (rel) {
+                    case "shortcut icon":
+                        iconType = .Icon
+                    case "icon":
+                        iconType = .Icon
+                    case "apple-touch-icon":
+                        iconType = .AppleIcon
+                    case "apple-touch-icon-precomposed":
+                        iconType = .AppleIconPrecomposed
+                    default:
+                        iconType = nil
                     }
+                }
 
-                    guard let href = link.attribute("href") where iconType != nil else {
-                        return
```

Comment:
Curious. Doesnt the return here mean that this would have exited out of the loop without returning any icons? If so how was this working?